### DISCUSSION TITLE
Inquiries about E1 Basic Hazus Example Template and Output

### DISCUSSION THREAD

**Discussion by MohamedKhoudari** [NHERI-SimCenter/SimCenterCommon#246] [R2D] [answered]
Dear R2D support team,

I am getting started with R2D and planning to use it as part of my thesis project for a regional analysis and impact assessment of an earthquake event on an area in my country Lebanon. I will be providing the event grid intensity measures and buildings dataset following the “E1-Basic Hazus” example input data files format (.csv). After watching some of the introductory videos, I performed a few experimental runs on the E1-Basic Hazus example to explore its output. However, I have some concerns regarding the outputs and the run process, and it would be much appreciated if I had your support.

My questions are as follows:

1-	In the E1-Basic Hazus example, the RES tab only shows the regional map and a table including the Most Likely Critical Damage State, mean repair cost, mean repair time, std. repair cost, and std. repair time as indicated in the screenshot below.
![1](https://github.com/user-attachments/assets/f9a73397-64bc-4fc7-8025-53cfa34e47a0)
 
However, I am interested in getting (i) the probability of being in a specific damage state and (ii) the expected number of injuries/casualties in each building. The Developer Manual (section 2.6.2) indicates the different forms of output to be obtained after running a scenario (as shown in red in the screenshots below); however, when running the E1-Basic Hazus example I’m not getting any of these on the user interface RES tab. Could you please provide me with guidance on how to reach and obtain such results?
 
![2](https://github.com/user-attachments/assets/2d9c4be6-7525-4c16-81a2-8341ab726301)
![3](https://github.com/user-attachments/assets/511ed338-66f6-48ba-86a5-1453ab639ac0)

2-	Before uploading the E1 Basic Hazus example, the Results tab in the upper ribbon displays options including Casualties, Detailed results, etc.… while when uploading E1 those options become limited to Building and Regional Map only as indicated below.
![4](https://github.com/user-attachments/assets/e627732a-3592-4a3b-8c2b-ff6ab3828bdc)

![5](https://github.com/user-attachments/assets/7fc02226-30db-466d-84b1-d474ae9ff1c6)

Is this a limitation on the E1 Basic Hazus example template that is also causing the problem of not getting the mentioned results in my previous question?

3-	I tried to start a run without uploading E1 as an example. I copied the input data files of E1 and placed them on my desktop. Then, without uploading E1, I filled in all the needed files' paths in HAZ and ASD tabs, and selected the same methods and choices in all other tabs as in E1. However, I got an error. But when uploading E1 as a template, and replacing the example’s files' paths with paths to the mentioned copy of the input files, the run was successful.
Based on this, when I want to run a scenario on the region I am studying, should I upload E1 (or other example) as a template, and replace its Event and Buildings file paths with the paths to the files specific to my region of study?

4-   As part of my project, I am also interested in analyzing the effect of buildings damage on the transportation network and the traffic status. In the E14 Basic Hazus transportation example, is the effect of the nearby buildings' damage (debris generated by the buildings) taken into account in calculating the damage state of the road? And does the feature of conducting traffic simulation become available in R2D?

Looking forward to hearing from you. Your support and guidance would be much appreciated!

Best regards,
Mohamed Khoudari

**Comment by zsarnoczay** [comment, marked as answer]
Dear Mohamed,

Thank you for your interest in using R2D and for taking the time to describe your questions in detail.

I will be able to answer some of your questions and I ask my colleague @jinyan1214 to help me cover the rest of them.

1) Unfortunately, the data schema described in the documentation is outdated. We recently introduced a new schema to support a broader set of probabilistic models, and we are in the process of updating the documentation to reflect that schema. R2D intentionally only focuses on the high-level results to allow you to sanity-check your simulations. If you navigate to the working directory where the simulations ran, under the `tmp.SimCenter/Results/` folder you'll find JSON files that capture every single realization of damage for every building: 

- The `Results_det.json` file includes deterministic information for every asset in your inventory. This typically covers the asset location and its main features.
- The `Results_###.json` files describe the probabilistic information, where ### stands for the id of the realization. These files typically include the IMs, EDPs, Damage States, and Decision Variables.
- The `R2D_results.geojson` is a GeoJSON file that includes the high-level summary results that are shown in R2D.

If you are interested in the probability of being in a specific damage state for each building, you'll need to collect the damage state for that building in the various realizations and then calculate the relevant statistic from that information. Note that by having access to every realization, you'll also e able to consider more advanced statistics, such as correlations between damage states of different buildings.

We do not calculate injuries automatically because we were not happy with the basic model implementation in previous versions. We are working on an improved model now and removed the injury/casualty feature until the new model is available. If you are interested in Hazus-based injury estimates, you can calculate such injuries based on the damage state data in a relatively simple post-processing step. Let me know if you need help with that, I am happy to provide you guidance.

2) Before loading a particular set of results, the menu shows the complete list of options. As soon as you load the results, the options get limited to the type of results that are actually available. As I mentioned above, we do not calculate injuries in the current version, hence you do not see that option once you've loaded the example data.

3) There should be no difference between loading an example and setting up the simulations manually. Please share more information about the error message you see when trying to run the example manually so that we can help you identify which setting did not match the example in your manual configuration.

4) Debris is currently not taken into consideration, but it is something we could add in the future. We can estimate the amount of debris from building damage using the Hazus earthquake methodology, but deciding how that debris can block roads is another step that is not trivial. Please let us know if you have any particular methodology in mind when it comes to interaction between debris and roads.

If you have further questions, please do not hesitate to ask.

**Reply by MohamedKhoudari** [reply]
Dear Adam,
Thank you for your response. The guidance you provided is really helpful!!

Kindly find attached the R2D run error I got after manually setting up the input of example 1.
Regarding casualty estimation, it would be much appreciated if you can provide any resources or guidance for proceeding with this post-processing step.
As for the debris extent around a damaged building, I am still reviewing several methodologies and will get back to you once I've decided which one to adopt.

Thank you again for your help and support!
[R2D run error.docx](https://github.com/user-attachments/files/17612334/R2D.run.error.docx)

Best regards,
Mohamed Khoudari

**Comment by MohamedKhoudari** [comment]
Dear Adam and Jinyan,
I hope you are doing well.

I just wanted to remind you of the R2D run error attached above. It would be much appreciated if you can assist me in resolving it. This error is appearing after manually setting up the inputs of E1 in the user interface tabs and running the case. 
[R2D.run.error.docx](https://github.com/user-attachments/files/17792408/R2D.run.error.docx)

Best regards,
Mohamed Khoudari

**Reply by jinyan1214** [reply]
Hi Mohamed, 

Thanks for reminding and I apologize for the late response. 

By looking at the run error you attached, it seems you did not define units in the GI step. The units defined in the GI describe the units of the attributes in your asset inventory. Can you try defining the units and let us know if the error can be resolved?

Best,

Jinyan

**Reply by MohamedKhoudari** [reply]
Dear Jinyan,

Thank you for your feedback. I checked that the units are defined and the error was resolved.

One more thing I would like to inquire about is the methodology used in calculating the damage state (0, 1, 2, 3, or 4) for each realization. After plotting the fragility curve of the structure type under consideration and getting the intensity measure (PGA) value selected for a specific realization, should I use the expected damage state E[DS] formula (attached below) to get the same damage state generated by R2D, or does R2D use another approach to generate such values?
![image](https://github.com/user-attachments/assets/52f5fb75-0569-472c-a9a4-60fcb9578cad)

Thank you again for your continuous support!

Best regards,
Mohamed Khoudari

**Reply by jinyan1214** [reply]
Hi Mohamed,

R2D uses a package Pelicun to estimate the damage state, and Pelicun uses a Monte Carlo method to estimate the damage states. In R2D, you define your hazard (ground motion grid in E1), define hazard to asset mapping (nearest neighbor sampling in E1), and structural analysis method (IMasEDP in E1), and these will give you a list of  EDP (in E1 this is just the IM but can be real EDPs in other examples) realizations. These EDP realizations are created for each building and saved in "[path to your local work directory, by default your documents directory]/R2D/LocalWorkDir/tmp.SimCenter/Results/Buildings/8000/response.csv". These response.csv files, along with several other parameters, are passed to Pelicun for the damage and loss analysis. 

Once Pelicun has the EDP realizations, it will generate damage state realizations according to the conditional probabilities defined in the fragility curves. To explain this process in a simplified way, in the DL panel of R2D, a Sample size (e.g., 100) is defined. Pelicun will pick the first EDP realization from response.csv, generate one capacity realization for each damage state according to the fragility curves, and compare the EDP realization with the capacity realization. If the EDP realization is bigger than the capacity of DS_i, then a DS_i is reached. Pelicun will then go to the next EDP realization in response.csv and repeat the above process until the Sample size defined in DL panel is reached. Sometimes, the sample size in DL is bigger than the number of realizations in response.csv. In this case, when the EDP realization is used up, pelicun will start a new loop from the first EDP realization. The generated DL realizations are saved in "[path to your local work directory, by default your documents directory]/R2D/LocalWorkDir/tmp.SimCenter/Results/Buildings/8000/DMG_grp.json". In this json file, there will be several keys. The key corresponds to the fragility curves that are used to analyze this specific building. Plus, there is a key named "collapse". As defined in HAZUS manual, a building has a certain probability of collapsing when the DS is complete. This collapsing state is also sampled and saved under the "collapse" key.

The above is a simplified explanation of how Pelicun works, and there are some 'tricks' inside pelicun to calculate things in batches for efficiency. For more details, you can refer to [Pelicun documentation](https://nheri-simcenter.github.io/pelicun/user_guide/pelicun_framework.html). 

Hope this helps.

Best,

Jinyan
